files/system/usr/share/ublue-os/just/70-secureblue.just

```diff
@@ -534,28 +534,34 @@ audit-secureblue:
         for f in ${!flatpaks[@]}; do
             has_network=false
             has_x11=false
+            has_hardened_malloc=true
+            status=$STATUS_SUCCESS
             fullref=${flatpaks["$f"]}
             permissions=$(flatpak info --show-permissions "$fullref")
             if hasPermission "$permissions" "shared" "network"; then
                 has_network=true
+                [[ ! "$status" != "$STATUS_FAILURE" ]] && status="$STATUS_WARNING"
             fi
-            if hasPermission "$permissions" "sockets" "x11" && ! hasPermission "$permissions" "sockets" "fallback-x11"  ]]; then
+            if hasPermission "$permissions" "sockets" "x11" && ! hasPermission "$permissions" "sockets" "fallback-x11"; then
                 has_x11=true
+                status="$STATUS_FAILURE"
             fi
-            flatpak_test_string="Auditing $f"
-            if [[ ! $has_network == "true" && ! $has_x11 == "true" ]]; then
-                print_status "$flatpak_test_string" "$STATUS_SUCCESS"
-            elif [[ $has_x11 == "true" ]]; then
-                print_status "$flatpak_test_string" "$STATUS_FAILURE"
-            elif [[ $has_network == "true" ]]; then
-                print_status "$flatpak_test_string" "$STATUS_WARNING"
+            if ! hasPermission "$permissions" "LD_PRELOAD" "/var/run/host/usr/lib64/libhardened_malloc.so"; then
```

Comment:
this won't work because by default we use the optimized library

Reply:
I see. Should the script check specifically for matching architecture, or just any hardened_malloc?

Reply:
just the string "libhardened_malloc" is probably sufficient